microprofile/messaging/src/test/java/io/helidon/microprofile/messaging/AbstractCDITest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.messaging;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.logging.LogManager;
+import java.util.stream.Collectors;
+
+import javax.enterprise.inject.se.SeContainer;
+import javax.enterprise.inject.se.SeContainerInitializer;
+import javax.enterprise.inject.spi.CDI;
+
+import io.helidon.config.Config;
+import io.helidon.config.ConfigSources;
+import io.helidon.microprofile.server.Server;
+import io.helidon.microprofile.server.ServerCdiExtension;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+public abstract class AbstractCDITest {
+
+    static Server singleServerReference;
+
+    static {
+        try (InputStream is = AbstractCDITest.class.getResourceAsStream("/logging.properties")) {
+            LogManager.getLogManager().readConfiguration(is);
+        } catch (IOException e) {
+            fail(e);
+        }
+    }
+
+    protected SeContainer cdiContainer;
+
+    protected Map<String, String> cdiConfig() {
+        return Collections.emptyMap();
+    }
+
+    protected void cdiBeanClasses(Set<Class<?>> classes) {
+        //noop
+    }
+
+    @BeforeEach
+    public void setUp() {
+        Set<Class<?>> classes = new HashSet<>();
+        cdiBeanClasses(classes);
+        Map<String, String> p = new HashMap<>(cdiConfig());
+        cdiContainer = startCdiContainer(p, classes);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        try {
+            singleServerReference.stop();
+            cdiContainer.close();
+        } catch (Throwable t) {
+            //emergency cleanup see #1446
+            stopCdiContainer();
+        }
+    }
```

Comment:
@tomas-langer Tear down is double checked to avoid cross test pollution related to #1446